### PR TITLE
Fixed Stream decode C0 byte (nil) processing error

### PR DESCRIPTION
**Pull request by i-kovalyov**
Fixed Stream decode processing error:
after C0 byte (nil) stream switched to "End of stream state",
added test.
Minor changes made in modules formatting to satisfy modules
formatting rules checked in tests.

### CODE REVIEW

File: lib/streams.js [hunk 1]
```diff
@@ -81,5 +84,13 @@ Decoder.prototype._transform = function (buf, enc, done) {
   }
 }
 
+Decoder.prototype.emit = function () {
+  if ((arguments.length === 2) && (arguments[0] === 'data') && (arguments[1] === undefined)) {
+    arguments[1] = null
+  }
+
+  Base.prototype.emit.apply(this, arguments)
+}
```
Comment (mcollina) [inline lib/streams.js]: This is a very bad practice, and it will only work for on(‘data’) handlers and not piped streams.

File: test/streams.js [hunk 2]
```diff
@@ -182,3 +182,24 @@ test('concatenated buffers work', function (t) {
 
   encoder.end()
 })
+
+test('nil processing works', function (t) {
+  t.plan(3)
+
+  var pack = msgpack()
+  var decoder = pack.decoder()
+  decoder.wrap = true
```
Comment (mcollina) [inline test/streams.js]: can you pass this as an option instead?

**Comment by mcollina** [comment]
Can you also add the same for the encoder?

**Comment by i-kovalyov** [comment]
Sorry, to confirm for encoder.
Do you want to add "unwrap" option for encoder?
Using unwrap  = true encoder can
encode obj.value instead of obj.

Using this option we can pass decoder output to encoder.

**Comment by mcollina** [comment]
The overall goal is to have source.pipe(encoder).pipe(decoder).pipe(sink) have the same input/output sata. I would keep the name of the option the same, but do what make sense to you.

**Comment by mcollina** [comment]
What I mean is that the encode takes objects in the form of `{ value }`.

**Pull request merged**